code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -49,149 +50,177 @@ LINEN BINS
 
 /obj/item/weapon/bedsheet/green
 	icon_state = "sheetgreen"
 	item_color = "green"
-	
+	dream_messages = list("green")
+
 /obj/item/weapon/bedsheet/grey
 	icon_state = "sheetgrey"
 	item_color = "grey"
+	dream_messages = list("grey")
 
 /obj/item/weapon/bedsheet/orange
 	icon_state = "sheetorange"
 	item_color = "orange"
+	dream_messages = list("orange")
 
 /obj/item/weapon/bedsheet/purple
 	icon_state = "sheetpurple"
 	item_color = "purple"
+	dream_messages = list("purple")
 
 /obj/item/weapon/bedsheet/patriot
 	name = "patriotic bedsheet"
 	desc = "You've never felt more free than when sleeping on this."
 	icon_state = "sheetUSA"
 	item_color = "sheetUSA"
+	dream_messages = list("America", "freedom", "fireworks", "bald eagles")
 
 /obj/item/weapon/bedsheet/rainbow
 	name = "rainbow bedsheet"
 	desc = "A multicolored blanket. It's actually several different sheets cut up and sewn together."
 	icon_state = "sheetrainbow"
 	item_color = "rainbow"
+	dream_messages = list("red", "orange", "yellow", "green", "blue", "purple", "a rainbow")
 
 /obj/item/weapon/bedsheet/red
 	icon_state = "sheetred"
 	item_color = "red"
+	dream_messages = list("red")
 
 /obj/item/weapon/bedsheet/yellow
 	icon_state = "sheetyellow"
 	item_color = "yellow"
+	dream_messages = list("yellow")
 
 /obj/item/weapon/bedsheet/mime
 	name = "mime's blanket"
 	desc = "A very soothing striped blanket.  All the noise just seems to fade out when you're under the covers in this."
 	icon_state = "sheetmime"
 	item_color = "mime"
+	dream_messages = list("silence", "gestures", "a pale face", "a gaping mouth", "the mime")
 
 /obj/item/weapon/bedsheet/clown
 	name = "clown's blanket"
 	desc = "A rainbow blanket with a clown mask woven in. It smells faintly of bananas."
 	icon_state = "sheetclown"
 	item_color = "clown"
+	dream_messages = list("honk", "laughter", "a prank", "a joke", "a smiling face", "the clown")
 
 /obj/item/weapon/bedsheet/captain
 	name = "captain's bedsheet"
 	desc = "It has a Nanotrasen symbol on it, and was woven with a revolutionary new kind of thread guaranteed to have 0.01% permeability for most non-chemical substances, popular among most modern captains."
 	icon_state = "sheetcaptain"
 	item_color = "captain"
+	dream_messages = list("authority", "a golden ID", "sunglasses", "a green disc", "an antique gun", "the captain")
 
 /obj/item/weapon/bedsheet/rd
 	name = "research director's bedsheet"
 	desc = "It appears to have a beaker emblem, and is made out of fire-resistant material, although it probably won't protect you in the event of fires you're familiar with every day."
 	icon_state = "sheetrd"
 	item_color = "director"
+	dream_messages = list("authority", "a silvery ID", "a bomb", "a mech", "a facehugger", "maniacal laughter", "the research director")
 
 // for Free Golems.
 /obj/item/weapon/bedsheet/rd/royal_cape
 	name = "Royal Cape of the Liberator"
 	desc = "Majestic."
+	dream_messages = list("mining", "stone", "a golem", "freedom")

```

Comment:
I feel you should add the liberator's magic words of "Yeah, sure go do whatever."